scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala

```diff
@@ -421,13 +431,15 @@ class AstNamerMacros(val c: Context) extends Reflection with CommonNamerMacros {
           )
         }
 
-        mstats1 += q"""
-          object internal { // to be ignored by Mima
-            object Latest {
-              ..$mstatsLatest
-            }
-          }
-        """
+        val latestTermName = mstatsPerVersion.foldLeft(TermName(initialName)) {
+          case (afterPrevVer, (ver, verMstats)) =>
+            mstats1 += q"object $afterPrevVer { ..$verMstats }"
+            mstats1 += q"final val ${getUntilVersion(ver)} = $afterPrevVer"
+            getAfterVersion(ver)
+        }
+        mstats1 += q"object $latestTermName { ..$mstatsLatest }"
+        // to be ignored by Mima, use "internal"
+        mstats1 += q"object internal { final val Latest = $latestTermName }"
```

Comment:
Quasiquotes expanded by 4.6.[0-3] depend on this, so it would be nice to have some MiMa to honor "Every Scala program linkable against x.y.0 must be linkable against x.{y+1}.z."? That won't be an issue for Scalafix as it didn't ship with 4.6.[0-3], but that might bite other downstream projects.

Maybe it can be deprecated, frozen to `After_4_6_0` (or the historical `unapply` if it does not exist).

Reply:
The whole point of `internal` was that it was not subject to mima (and it was named `Impl` before and has already been renamed anyway). The leak via quasiquotes was unintentional, but I'd like to keep it this way, and incur minor incompatibility if anyone has already used it.

Reply:
Fair enough, especially since I guess the upcoming release will remain in the 4.7.x serie, so there is absolutely no reason for clients to stick to the earlier 4.7.x versions with an non-guaranteed compatibility.